src/ui/map.js

```diff
@@ -2171,22 +2171,58 @@ class Map extends Camera {
+     * Queries the currently loaded data for elevation at a geographical location. The elevation is returned in `meters` and accounts for the value of `exaggeration` set on `terrain`.
      * Returns `null` if `terrain` is disabled or if terrain data for the location hasn't been loaded yet.
      *
      * In order to guarantee that the terrain data is loaded ensure that the geographical location is visible and wait for the `idle` event to occur.
-     * @param {LngLatLike} lnglat The geographical location to project.
+     * @param {LngLatLike} lnglat The geographical location at which to query.
      * @returns {number | null} The elevation in meters, accounting for `terrain.exaggeration`.
      * @example
      * var coordinate = [-122.420679, 37.772537];
-     * var elevation = map.queryTerrainElevationAt(coordinate);
+     * var elevation = map.queryTerrainElevationAtLocation(coordinate);
      */
-    queryTerrainElevationAt(lnglat: LngLatLike): number | null {
+    queryTerrainElevationAtLocation(lnglat: LngLatLike): number | null {
         if (!this.transform.elevation) return null;
 
         return this.transform.elevation.getAtPoint(MercatorCoordinate.fromLngLat(lnglat));
     }
 
+    /**
+     * Queries the visible data for elevation at a geographical point that corresponds to the specified pixel coordinates. The elevation is returned in `meters` and accounts for the value of `exaggeration` set on `terrain`.
+     * Returns `null` if `terrain` is disabled, if terrain data for the location hasn't been loaded yet, or when the specified point is above the horizon.
+     *
+     * In order to guarantee that terrain data is loaded wait for the `idle` event to occur.
+     * @param {PointLike} point The pixel coordinates at which to query.
+     * @returns {number | null} The elevation in meters, accounting for `terrain.exaggeration`.
+     * @example
+     * map.on('mousemove', function(e) {
+     *   // Get the elevation underneath the mouse pointer
+     *   var elevation = map.queryTerrainElevation(e.point);
+     * });
+     */
+    queryTerrainElevation(point: PointLike): number | null {
+        if (!this.transform.elevation) return null;
+
+        const result = this.transform.elevation.pointCoordinate(Point.convert(point));
+        return result ? result[3] : null;
+    }
+
+    /**
+     * Returns `true` when the pixel underneath `point` is a part of the Map
+     * and `false` when the pixel underneath is part of the surrounding whitespace or a `sky` layer.
+     *
+     * @param {PointLike} point The pixel coordinates for a point in the container.
+     * @returns {boolean} `true` when `point` is over the Map.
+     * @example
+     * map.on('mousemove', function(e) {
+     *   // Check if the mouse is over the Map
+     *   var onMap = map.isUnderneath(e.point);
+     * });
+     */
+    isUnderneath(point: PointLike): boolean {
+        return this.transform.isPointOnMap(Point.convert(point));
```

Comment:
Concerning the behavior of this API:
- Is there any overlap between this and the internal: `transform.isPointAboveHorizon()`? Basically, is `!transform.isPointAboveHorizon(point)` equivalent to `isPointOnMap(point)`?
- As it seems like this API does not account for extrusion above the map plane, should we give a mention of that in the docs?

Reply:
I added a comment for 1. 
https://github.com/mapbox/mapbox-gl-js/pull/10602/files#r617818148

Good point about extrusions. Would it be clearer if we renamed the function to `isOverGround` ? or something similar.

Reply:
How about simply `map.isOver(Point)`?

Reply:
oh yea, how about `isOverlapping` ? because over has weird height connotations (floating over)

Reply:
Ok how about even simpler `map.overlaps(point)`?

Reply:
I like `overlaps` because it's really succinct and clear, but using a verb doesn't seem common in our APIs (haven't seen any), it seems like all checks are using the `is*` approach, so maybe what you suggested (`isOverlapping`?),

Reply:
> **Overlap**: To lie or extend over and cover part of.

Overlaps what?

Similar to https://github.com/mapbox/mapbox-gl-js/pull/10602#discussion_r617205950, Can this be folded into `unproject` with an option or as an overload? If it's too soon to do that, maybe we don't add this method yet.

Reply:
On terrain, if there is a use case to get point on map, if it raycast hits the map, this is not optimal for performance point of view: this would mandate two DEM tree raycast for each point: unproject & overlap.

How about adding method `raycast` - that would not clamp like `unproject` and provide elevation for the raycast point and null, if map is not hit? 
I understand that `raycast` already sounds like a convenience wrap to queryRenderedFeatures. I'm not worried about that though - in a way, `overlaps` here could be seen as implementing `(map.queryRenderedFeatures(e.point, {layers:['sky']}).length == 0)`.